util/trace_replay.cc

```diff
@@ -127,12 +127,19 @@ Status Tracer::WriteFooter() {
 }
 
 Status Tracer::WriteTrace(const Trace& trace) {
-  std::string encoded_trace;
-  PutFixed64(&encoded_trace, trace.ts);
-  encoded_trace.push_back(trace.type);
-  PutFixed32(&encoded_trace, static_cast<uint32_t>(trace.payload.size()));
-  encoded_trace.append(trace.payload);
-  return trace_writer_->Write(Slice(encoded_trace));
+  if (trace_options_.trace_filter_option == TraceFilterType::kTraceFilterNone
```

Comment:
Instead of modifying `WriteTrace`, could you add this filtering logic in `ShouldSkipTrace` method?